src/components/Signup/Signup.component.tsx

```diff
@@ -0,0 +1,156 @@
+                ? 'Vos deux mots de passe doivent être identiques'
+                : ''
+        });
+    }, [userInput]);
+
+    function onSignupSubmit (): void {
+        if (isFormValid()) {
+            signupSubmit({
+                variables: {
+                    email,
+                    password
+                }
+            });
+        }
+        setTriggerValidation(true);
+    }
+
+    function onPopupClose () {
+        setShouldDisplayPopupConfirm(false);
+        history.push(Routes.LOGIN);
+    }
+
+    return (
+        <>
+            <div className="login">
+                <div className="login__card">
+                    <img className="beaver-logo" src={beaver} alt="" />
+                    <Form classname="signup__form">
+                        <h1 className='signup__title'>S&apos;inscrire</h1>
+                        <InputForm
+                            name="nickname"
+                            value={nickname}
+                            placeholder="nickname"
+                            isRequired
+                            onChange={(event) => onInputChange(event, setUserInput, userInput)}
+                            type='text'
+                        />
+                        {triggerValidation && nicknameValidation && <p className="form-error">{nicknameValidation}</p>}
+                        <InputForm
+                            name="email"
+                            value={email}
+                            placeholder="Email"
+                            isRequired
+                            onChange={(event) => onInputChange(event, setUserInput, userInput)}
+                            type='email'
+                        />
+                        {triggerValidation && emailValidation && <p className="form-error">{emailValidation}</p>}
+                        <InputForm
+                            name="password"
+                            value={password}
+                            placeholder="Mot de passe"
+                            isRequired
+                            type="password"
+                            onChange={(event) => onInputChange(event, setUserInput, userInput)}
+                        />
+                        {triggerValidation && passwordValidation && <p className="form-error">{passwordValidation}</p>}
+                        <InputForm
+                            name="confirmPassword"
+                            value={confirmPassword}
+                            placeholder="Confirmation de mot de passe"
+                            isRequired
+                            type="password"
+                            onChange={(event) => onInputChange(event, setUserInput, userInput)}
+                        />
+                        {triggerValidation && confirmPasswordValidation && <p className="form-error">{confirmPasswordValidation}</p>}
+                        <Button
+                            width="100%"
```

Comment:
Question de gout perso, mais je prefere eviter de set du CSS dans le JS quand c'est possible, ici dans Signup.style.scss c'est possible de faire
```suggestion
.signup__form button {
    width: 100%
}
```

Reply:
On a une prop width au composant Button. Tu voudrais l'enlever du coup?